### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by ayan-b**

### CODE REVIEW

**Comment by ayan-b** [comment]
@yunhailuo  Submitted this PR to get quick feedback. :smiley:

**Comment by yunhailuo** [comment]
Maybe rebase to master since I just merged #12 ?

**Comment by ayan-b** [comment]
> Maybe rebase to master since I just merged #12 ?

Rebased. :)

**Comment by ayan-b** [comment]
@yunhailuo, I think `equal_matrices`, `merge_xena`, `TARGET-CCSK_phenotype_ETL` modules could be inside `utils.py`.

**Comment by ayan-b** [comment]
> @yunhailuo, I think `equal_matrices`, `merge_xena`, `TARGET-CCSK_phenotype_ETL` modules could be inside `utils.py`.

@yunhailuo What do you think about this? :)

**Comment by yunhailuo** [comment]
`equal_matrices` and `merge_xena` can go to utils. I said they can go into utils only because they can be used by other python scripts/modules as well. But I would feel better if they are moved together with your #16 because they are, at least back in days for me, used for comparing Xena matrices, meaning an CLI interface seems necessary.

`TARGET-CCSK_phenotype_ETL`, if I remember correctly, is a really special case which is less likely to be applicable to other python scripts/modules. For me, it's better to leave it as a single, isolated, less used script.

**Comment by yunhailuo** [comment]
BTW, I'm actually OK if you want to move one or a few functions at a time/PR, instead of doing everything perfect in one PR. The only recommendation is to make the PR message clear because I would, by default, do "Squash and merge" and use PR title as final commit title.

**Comment by ayan-b** [comment]
In this particular PR, instead of `Squash and merge`, only `Merge` can be done since there are movements of different functions from different files(`mkdir_p`, `make_metadata`). I will separate out the commits in that way before merging.

**Comment by yunhailuo** [comment]
> In this particular PR, instead of `Squash and merge`, only `Merge` can be done since there are movements of different functions from different files(`mkdir_p`, `make_metadata`). I will separate out the commits in that way before merging.
> 
> As a sidenote, pylint has a feature to check [code similarity](https://pylint.readthedocs.io/en/latest/technical_reference/features.html#similarities-checker). We can enable this in CI in the lint stage.

I would consider "Rebase and merge" first unless there is a strong reason for a merge commit. I don't want to deal with merge commits if I don't have to. Also, make sure you have a clean commit history if you don't want to have squash. I'll mark this PR as "Rebase and merge" for now.

**Comment by yunhailuo** [comment]
Just a mark for myself, 15993f5 looks nice and clean by itself.

**Comment by ayan-b** [comment]
@yunhailuo I think this PR is good to go. Let me know if some modification is required.

**Comment by ayan-b** [comment]
@yunhailuo Should I make more changes? I don't think `make_metadata` overlaps anymore with `xena_datasets` module.

**Comment by ayan-b** [comment]
> I'm not sure about your vision on make_metadata and xena_datasets. I just feel like it's better not to import scripts in a module.

Will follow.

**Comment by ayan-b** [comment]
@yunhailuo I have found that quite a few modules inside scripts (namely `make_metadata`, `merge_xena`, `panTCGA`) `meta_templates` and `metavars` have been used. However, they are private data members inside `xena_dataset.GDCOmicset`. I feel it would be good if we declare them in `xena_gdc_etl/__init__.py` itself.

**Comment by ayan-b** [comment]
Update on implementation: I have used stored the `METADATA_VARIABLES` and `METADATA_TEMPLATE` in a `constant.py` file instead of `__init__.py`.

**Comment by yunhailuo** [comment]
> @yunhailuo I have found that quite a few modules inside scripts (namely `make_metadata`, `merge_xena`, `panTCGA`) `meta_templates` and `metavars` have been used. However, they are private data members inside `xena_dataset.GDCOmicset`. I feel it would be good if we declare them in `xena_gdc_etl/__init__.py` itself. In this way, it will available to every module which imports `xena_gdc_etl`. What do you think?

Sounds good. Please feel free to do that.

**Comment by ayan-b** [comment]
It's already done. Instead of putting them inside `__init__.py`, I have put them in `constants.py`.

**Comment by ayan-b** [comment]
@yunhailuo I think this PR can be merged. Let me know if some changes are required.

**Comment by ayan-b** [comment]
> Hi. I see that the json module is imported but not used in tests/test_gdc.py.
It will be better if it's removed :)

Thanks for the review! Really appreciate it. :blush:

**Comment by ayan-b** [comment]
I have marked some tests as `CI` as they require internet while executing.

**Comment by yunhailuo** [comment]
> Note: I have marked some tests as `CI` as they require internet while executing. So, if someone is offline during coding, they can run `python -v -m "not CI"` in order to execute the tests which require no internet connection.

Just to check `python -v -m "CI"` is to execute tests requiring internet connection, right?

**Comment by ayan-b** [comment]
Yup.

**Comment by ayan-b** [comment]
@yunhailuo is the CONTRIBUTING guide OK or should remove/add something?

**Comment by yunhailuo** [comment]
> @yunhailuo is the CONTRIBUTING guide OK or should I remove/add something?

Should be fine for now for this PR. Is this PR ready?

**Comment by ayan-b** [comment]
This PR is ready and can be merged.

File: xena_gdc_etl/scripts/make_metadata.py [hunk 1]
```diff
@@ -18,6 +18,11 @@
 
 import jinja2
 
+sys.path.insert(
+    0, os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+)
+from xena_gdc_etl.constants import METADATA_TEMPLATE, METADATA_VARIABLES # noqa
+
```
Comment (yunhailuo) [inline xena_gdc_etl/scripts/make_metadata.py]: Looks strange here (and below). Relative import doesn't work here?

Comment (ayan-b) [reply xena_gdc_etl/scripts/make_metadata.py]: Relative import works here also. I used this method in order to have a similar import structure in all files (i.e., inside `xena_GDC_ETL/Scripts/` and `xena_GDC_ETL/`). I will change this to relative import.

**Comment by ayan-b** [comment]
> Since we are going to do rebase and merge for this PR, I'd appreciate it you could clean up the commit history a little bit. It seems ebb4970, 285da6b and 32ef0cc are dealing with one same issue. Should they be combined?

Sure.

**Comment by ayan-b** [comment]
I have created #22 for the test fix since it does not really belong in this PR.

**Comment by yunhailuo** [comment]
> I have created #22 for the test fix since it does not really belong in this PR.

That's good. Thanks again for the great work. I'll merge this one and see if #22 will pass.

**Pull request merged**